apps/journeys-admin/src/components/TemplateGallery/TemplateGallery.tsx

```diff
@@ -36,28 +35,18 @@ export function TemplateGallery(): ReactElement {
     void router.push(router)
   }
 
+  function handleLanguageChange(values: string[]): void {
+    setLanguageIds(values)
+    router.query.languageIds = values
+    void router.push(router)
+  }
+
   return (
     <Container disableGutters>
-      <Stack
-        gap={1}
-        justifyContent="space-between"
-        sx={{
-          pb: { xs: 6, md: 9 },
-          flexDirection: { xs: 'column', md: 'row' },
-          alignItems: { xs: 'start', md: 'center' }
-        }}
-      >
-        <Typography variant="h2" sx={{ display: { xs: 'none', lg: 'block' } }}>
-          {t('Journey Templates')}
-        </Typography>
-        <Typography variant="h4" sx={{ display: { xs: 'block', lg: 'none' } }}>
-          {t('Journey Templates')}
-        </Typography>
-        <LanguageFilter
-          languageId={languageId}
-          onChange={(value) => setLanguageId(value)}
-        />
-      </Stack>
+      <HeaderAndLanguageFilter
+        languageIds={languageIds}
```

Comment:
can we make this match the other component TagsFilter and make this prop:
```suggestion
        selectedLanguageIds={selectedLanguageIds}
```